app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -1,47 +1,261 @@
+    private fun setBatteryIcon(batteryLevel: String) {
+        val resource = when (batteryLevel.toInt()) {
+            in 81..100 -> R.drawable.ic_battery_100
+            in 61..80 -> R.drawable.ic_battery_80
+            in 41..60 -> R.drawable.ic_battery_60
+            in 21..40 -> R.drawable.ic_battery_40
+            in 3..20 -> R.drawable.ic_battery_20
+            in 0..3 -> R.drawable.ic_battery_0
+            else -> R.drawable.ic_battery_0
         }
+        binding.scooterInfo.batteryIndicatorIV.setImageResource(resource)
     }
 
+    @SuppressLint("PotentialBehaviorOverride")
+    override fun onMapReady(googleMap: GoogleMap) {
+        googleMap.setMapStyle(MapStyleOptions.loadRawResourceStyle(requireContext(), R.raw.map_style))
+        scooterMap = googleMap
+        scooterMap.moveCamera(CameraUpdateFactory.newLatLng(CLUJANGELES))
+        scooterMap.animateCamera(CameraUpdateFactory.newLatLngZoom(CLUJANGELES, ZOOM_LEVEL))
+        scooterMap.setOnCameraIdleListener {
+            Handler(Looper.getMainLooper()).postDelayed({
+                mapViewModel.currentLocation.value = scooterMap.cameraPosition.target
+                logTag("COORDS", mapViewModel.currentLocation.value.toString())
+            }, 5000L)
+        }
+        scooterMap.setOnMarkerClickListener {
+            initScooterInfo(it.title?: "0000")
+            return@setOnMarkerClickListener true
+        }
 
-    private fun initViews() {
-        binding.mapTV.text = authViewModel.userData.value?.user?.email
+        scooterMap.setOnMapClickListener {
+            binding.scooterInfo.scooterCV.visibility = View.GONE
+        }
+    }
+
+    private fun isLocationEnabled(): Boolean {
+        val locationManager: LocationManager =
+            requireActivity().getSystemService(LOCATION_SERVICE) as LocationManager
+        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER) || locationManager.isProviderEnabled(
+            LocationManager.NETWORK_PROVIDER
+        )
+    }
+
+    private fun checkPermissions(): Boolean {
+        if (ActivityCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED &&
+            ActivityCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+        ) {
+            return true
+        }
+        return false
+    }
+
+    private fun requestPermissions() {
+        ActivityCompat.requestPermissions(
+            requireActivity(),
+            arrayOf(
+                Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ),
+            permissionId
+        )
+    }
+
+    @SuppressLint("MissingSuperCall")
+    override fun onRequestPermissionsResult(
```

Comment:
This method is deprecated. Use contracts instead, as mentioned in the provided documentation
https://developer.android.com/training/location/permissions#user-choice-affects-permission-grants